src/audio/src/src_generic.c

```diff
@@ -167,23 +159,17 @@ static inline void fir_filter_generic(int32_t *rp, const void *cp, int32_t *wp0,
 		 * Qx.31.
 		 */
 		for (i = 0; i < n1; i++) {
-			y0 += (int64_t)(*coef >> 8) * (*data);
-			data++;
-			y1 += (int64_t)(*coef >> 8) * (*data);
-			data++;
-			coef++;
+			y0 += (int64_t)(*coef >> 8) * (*(data++));
+			y1 += (int64_t)(*(coef++) >> 8) * (*(data++));
```

Comment:
as you note this doesn't change performance, what might however indeed save an instruction or two is saving and reusing the result of `*coef >> 8`. Possibly the compiler does that already, then this won't improve anything. But if it doesn't, using a variable like `int32_t tmp = *coef >> 8;` (choose a better name) inside the loop might optimise this a bit

Reply:
@lyakh What's your opinion about declaring variable inside for loop? Good or bad? It might help optimizer but OTOH I see with default optimization in debugger <optimized out> for most variables when not near their use locations so the compiler should know when they need to exist.

Reply:
@singalsu @lyakh , code write with such format is due to align with DSP asm or intrinsic functions, when turn to intrinsic, the value you mentioned: *coef >> 8 already in a register, no need write tmp = *coef >> 8.

if you see hifi version, you will have no such concerns.

Reply:
> @lyakh What's your opinion about declaring variable inside for loop? Good or bad? It might help optimizer but OTOH I see with default optimization in debugger for most variables when not near their use locations so the compiler should know when they need to exist.

All proposed changes are optional, don't think the difference would be significant either way. Use the form that you feel most comfortable with.